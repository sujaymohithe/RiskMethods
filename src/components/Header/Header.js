import React from 'react';
import './Header.css';

const Header = (props) => {
  return (
    <header className="mainHeader">
      <div>
        <a href="/" data-test='Logo' className="logo">RISK METHODS</a>
      </div>
      {
        props.isAuthenticated &&
        <div className="divProfile">
          <p className="logo">Welcome {localStorage.userId && localStorage.userId.split('@')[0]}</p>
          <a href="/logout" className="logo signout">Logout</a>
        </div>
      }
    </header >
  );
}

export default Header;
