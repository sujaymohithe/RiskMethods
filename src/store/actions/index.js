export {
    auth,
    logout,
    setAuthRedirectPath,
    authCheckState
} from './auth';

export {
    InitIndicatorMessages,
    InitIndicatorMessageDetails
} from './indicatorMessage';