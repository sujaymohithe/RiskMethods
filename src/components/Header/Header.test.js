import React from 'react';
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import { store } from '../../store/store';
import Header from './Header';

Enzyme.configure({ adapter: new Adapter() });

describe('<Header/> component', () => {

    it('renders without parent component', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });

    const component = mount(
        <Provider store={store} >
            <Header isAuthenticated/>
        </Provider>);

    it('renders Header Component with logo', () => {
        const wrapper = component.find(Header).find(`[data-test='Logo']`);
        expect(wrapper.text() == "RISK METHODS");
    })

    it('Should render the logout and welcome sign if authenticated', () => {
        expect(component.find('.divProfile').length).toEqual(1);
    })

    const header = mount(
        <Provider store={store} >
            <Header />
        </Provider>);

    it('Should not render the logout and welcome sign if not authenticated', () => {
        expect(header.find('.divProfile').length).toEqual(0);
    })
})