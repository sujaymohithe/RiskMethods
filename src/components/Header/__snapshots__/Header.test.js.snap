// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> component renders without parent component 1`] = `ShallowWrapper {}`;

exports[`<HomeLayout/> Container renders without parent component 1`] = `ShallowWrapper {}`;
